src/Router.ts

```diff
@@ -35,15 +35,33 @@ export default class Router implements IRouter {
    // using the case-sensitivity setting of this router.
 
    public handle(originalErr: unknown, req: Request, resp: Response, done: NextCallback): void {
-      const processors = _.filter(this._processors, (p) => { return p.matches(req); });
-
-      const go = _.reduce(processors.reverse(), (next: NextCallback, p: IProcessorChain): NextCallback => {
-         return (err) => {
-            p.run(err, req, resp, next);
-         };
-      }, done);
-
-      go(originalErr);
+      const processors = this._processors;
+
+      let index = 0;
+
+      const processRequest = (err: unknown, processorReq: Request, processorResp: Response, next: NextCallback): void => {
+         let processor = processors[index];
+
+         index += 1;
+
+         if (processor === undefined) {
+            // We've looped through all available processors.
+            return next(err);
+            // User-defined route handlers may change the request object's URL to re-route
+            // the request to other route handlers. Therefore, we must re-check whether
+            // the current processor matches the request object after every processor is
+            // run.
```

Comment:
Mmm yeah, I honestly don't know where the right placement for this comment is. I thought maybe the because the indentation appeared _after_ the `return` statement and approximately lined up with the `else if` that it was clear enough that it applied to the `else if`, but I guess not if it doesn't read that way to you.

I didn't put it _after_ the `else if` because then to me it very clearly implies it's commenting on the next line, which it's not. So 🤷‍♂️ 

Which is the lesser of the two evils? If the indentation thing is too weird to you, then I'll just move it below. It almost needs a `^^^^^` on the first line of the comment 😆

Reply:
Yeah, this is an age-old conundrum. I slightly favor in the _block_ that it applies to. Fine adding a `^^^^` qualifier.